outlook/src/taskpane/components/ProfileCard/ProfileCard.tsx

```diff
@@ -28,6 +32,33 @@ export class ProfileCard extends React.Component<ProfileCardProps, {}> {
         this.state = {};
     }
 
+    private addDefaultSrc(ev) {
+        ev.target.src = defaultCompanyImageSrc;
+    }
+
+    private getIconOrInitials = () => {
+
+        if (this.props.parentIsCompany)
+        {
+            return null;
+        }
+
+        let iconOrInitials = <img className='icon' src={this.props.icon} onError={this.addDefaultSrc} />;
```

Comment:
Can't you do a cleaner if / elif / else with herebelow ``if (!this.props.icon)`` + ``if (this.props.domain)`` ? Current code has a lot of side-effects and in the end it is not easy to understand what will be the value of ``iconOrInitials``

Reply:
Code rearranged, you can now have a look.